### PR TITLE
Fixup override docs and validation

### PR DESCRIPTION
**Pull request by marun**

### CODE REVIEW

File: scripts/sync-up-helm-chart.sh [hunk 1]
```diff
@@ -27,6 +27,16 @@ CHART_FEDERATED_CRD_DIR="${CHART_FEDERATED_CRD_DIR:-charts/kubefed/charts/contro
 CHART_FEDERATED_PROPAGATION_DIR="${CHART_FEDERATED_PROPAGATION_DIR:-charts/kubefed/templates}"
 TEMP_CRDS_YAML="/tmp/kubefed-crds.yaml"
 
+OS=`uname`
+SED=sed
+if [ "$OS" == "Darwin" ];then
```
Comment (font) [inline scripts/sync-up-helm-chart.sh]: nit: s/$OS/${OS}/

Comment (marun) [reply scripts/sync-up-helm-chart.sh]: Done

**Comment by marun** [comment]
Addressed comment.

File: pkg/kubefedctl/enable/validation.go [hunk 2]
```diff
@@ -137,6 +138,11 @@ func federatedTypeValidationSchema(templateSchema map[string]v1beta1.JSONSchemaP
 												},
 											},
 										},
+										// TODO(marun) Enable defaulting for federated types
```
Comment (font) [inline pkg/kubefedctl/enable/validation.go]: I don't think enabling defaulting for federated types should constitute making `op` required, as there will be CRD client-side validation that would still fail. However, we can still make it required in validation while still leaving the field optional.

Comment (marun) [reply pkg/kubefedctl/enable/validation.go]: Fair point, I'll remove the comment. I find this behavior very counter-intuitive, and question whether client-side validation is more a blessing or a curse.

**Comment by marun** [comment]
Addressed comment.

**Comment by shashidharatd** [comment]
Thanks for taking it up.
/lgtm

**Pull request merged**